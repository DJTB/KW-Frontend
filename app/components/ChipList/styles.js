import styled from 'styled-components';
import { resetList } from 'shared/styles/utils';

export const Ul = styled.ul`
  ${resetList}
  /* TODO: any particular chiplist css style? */
`;
