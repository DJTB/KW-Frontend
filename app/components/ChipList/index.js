import React from 'react';
import PropTypes from 'prop-types';
import cuid from 'cuid';

import Chip from 'components/Chip';
import { Ul } from './styles';

ChipList.propTypes = {
  chips: PropTypes.arrayOf(PropTypes.string).isRequired,
  componentProps: PropTypes.object,
};

ChipList.defaultProps = {
  componentProps: {},
};

function ChipList({ chips, componentProps }) {
  return (
    <Ul>
      {chips.map((text) => <Chip key={cuid()} {...componentProps} >{text}</Chip>)}
    </Ul>
  );
}

export default ChipList;
