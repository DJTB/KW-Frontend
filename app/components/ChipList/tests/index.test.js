import React from 'react';
import { shallow } from 'enzyme';

import ChipList from '../index';

describe('<ChipList />', () => {
  it('should match baseline snapshot', () => {
    const renderedComponent = shallow(
      <ChipList chips={['chip1', 'chip2']} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should adopt component props', () => {
    const renderedComponent = shallow(
      <ChipList componentProps={{ color: 'purple', bgColor: 'whiteLight' }} chips={['chip1', 'chip2']} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
