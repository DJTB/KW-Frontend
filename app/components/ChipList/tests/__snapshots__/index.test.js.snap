// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChipList /> should adopt component props 1`] = `
<styled.ul>
  <Chip
    bgColor="white"
    textColor="purple"
  >
    chip1
  </Chip>
  <Chip
    bgColor="white"
    textColor="purple"
  >
    chip2
  </Chip>
</styled.ul>
`;

exports[`<ChipList /> should match baseline snapshot 1`] = `
<styled.ul>
  <Chip
    bgColor="white"
    textColor="blackLight"
  >
    chip1
  </Chip>
  <Chip
    bgColor="white"
    textColor="blackLight"
  >
    chip2
  </Chip>
</styled.ul>
`;
